src/pages/LevelsPage.js

```diff
@@ -0,0 +1,145 @@
+    }
+
+    // Base number of words per level
+    const baseWordsPerLevel = Math.floor(totalWords / levelCount);
+
+    // Extra words that need to be added
+    const extraWords = totalWords % levelCount;
+
+    // Initialize levels
+    const levels = Array.from({ length: levelCount }, () => []);
+
+    let wordIndex = 0;
+
+    // Distribute the base number of words to each level
+    for (let i = 0; i < levelCount; i++) {
+      levels[i] = words.slice(wordIndex, wordIndex + baseWordsPerLevel);
+      wordIndex += baseWordsPerLevel;
+    }
+
+    // Distribute the extra words starting from the first level towards the end
+    for (let i = 0; i < extraWords; i++) {
+      levels[i].push(words[wordIndex]);
+      wordIndex++;
+    }
+
+    return levels;
+  };
+
+  useEffect(() => {
+    const fetchLevels = async () => {
+      try {
+        const path = await getPathByName(pathName);
+        if (!path) {
+          setError('Path not found');
+          return;
+        }
+
+        const fetchedWords = await getWordsForPath(path.id);
+        if (!fetchedWords || fetchedWords.length === 0) {
+          setError('No words found for this path');
+          return;
+        }
+
+        const levels = wordsIntoLevels(fetchedWords);
+        setLevels(levels);
+        setLoading(false);
+      } catch (error) {
+        setError('Error fetching path or levels');
+        console.error(error);
+        setLoading(false);
+      }
+    };
+
+    fetchLevels();
+  }, [pathName]);
+
+  if (loading) return <p>Loading levels...</p>;
+  if (error) return <p>{error}</p>;
+
+  return (
+    <div className="levels-page">
+      <div className="levels-top-bar">
+        <BackButton className="levels-back-button" />
+      </div>
+      <div className="path-container">
+        <Path className="path" />
+        {levels.map((level, index) => {
+          const ButtonComponent =
+            index === levels.length - 1
+              ? finalButton
+              : levelButtons[Math.floor(Math.random() * levelButtons.length)];
+
+          const position = positions[index];
+
+          return (
+            <ButtonComponent
+              key={index}
+              onClick={() => navigate(`/peli/${pathName}/taso/${index + 1}`)}
+              style={{
+                position: 'absolute',
+                top: `${position[0]}px`,
+                left: `${position[1]}px`,
+                zIndex: '1',
+                width: '131.579px',
+                height: '141.228px',
+                borderRadius: '50%',
+                display: 'flex',
+                alignItems: 'center',
+                justifyContent: 'center',
+                fontSize: '60px',
+                background: '#72A895',
+                color: '#06705b',
+                border: 'none',
+                boxShadow: '0 10px 0 rgba(0, 0, 0, 0.3)',
+                transition: 'box-shadow 0.3s ease',
+                cursor: 'pointer',
```

Comment:
Miksi tämä on upotettu JS tiedostoon, eikä ole css tiedostona?

Reply:
Tämä on JS-tiedostossa sen takia, että haluan sijoittaa napit aina tiettyyn kohtaan polku-vektoria ja tämä sijainti perustuu nappia vastaavan tason indeksiin eli esim. ensimmäisen tason napilla on eri sijainti kuin toisen, kolmannen ja neljännen jne. Lisäksi napin väri pitäisi jatkossa määrittyä tason statuksella (onko se suoritettu, lukittu vai avoin suoritettavaksi). Koitin toteuttaa tämän niin, että suurin osa tyylistä olisi css-tiedostossa, ja ainoastaan tuo sijainti määritettäisiin tässä JS-tiedostossa, mutta ainoastaan "tarkemmin" määritetty tyyli oikeasti vaikuttaa tällöin nappiin, joten se ei onnistunut. En tiedä voisiko tämän sijainnin ja värin muuttamisen napin/tason indeksin perusteella toteuttaa jotenkin siellä css-tiedostossa? Otan tähän mielellään vinkkejä vastaan, koska olisihan tämä siistimpi, jos kaikki tyylit ovat samassa paikassa.

Reply:
En siis itse tiedä sen paremmin, katsotaan vaikka miitissä tarkemmin. Sekin voi olla ihan ok että nuo css jutut ovat suoraan tuossa, dynaamiset arvot ovat ihan hyvä peruste sille.